### PR TITLE
refactor SqlCommandParser

### PR DESCRIPTION
**Pull request by godfreyhe**
only `show current_catalog`, `show current_database`, `show modules`, `set`, `reset`, `describe`, `explain` are parsed through regex matching, other commands are parsed through sql parser.

### CODE REVIEW

File: src/main/java/com/ververica/flink/table/gateway/SessionManager.java [hunk 1]
```diff
@@ -26,6 +26,9 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
```
Comment (TsReaper) [inline src/main/java/com/ververica/flink/table/gateway/SessionManager.java]: Unused import?

File: src/main/java/com/ververica/flink/table/gateway/SqlCommandParser.java [hunk 2]
```diff
@@ -27,35 +51,142 @@
...
 
 	private SqlCommandParser() {
 		// private
 	}
 
-	public static Optional<SqlCommandCall> parse(String stmt) {
+	/**
+	 * Parse the given statement and return corresponding SqlCommandCall.
+	 *
+	 * <p>only `show current_catalog`, `show current_database`, `show modules`, `set`, `reset`,
```
Comment (TsReaper) [inline src/main/java/com/ververica/flink/table/gateway/SqlCommandParser.java]: show current catalog, show current database (without underscore)

Comment (godfreyhe) [reply src/main/java/com/ververica/flink/table/gateway/SqlCommandParser.java]: flink sql parser does not support `show current catalog` and `show current database`

**Comment by godfreyhe** [comment]
merging

**Pull request merged**